presto-kudu/src/main/java/io/prestosql/plugin/kudu/KuduSplitManager.java

```diff
@@ -45,10 +60,26 @@ public ConnectorSplitSource getSplits(
             ConnectorTableHandle table,
             SplitSchedulingStrategy splitSchedulingStrategy,
             DynamicFilter dynamicFilter)
+    {
+        if (dynamicFilter.isComplete() || !dynamicFilter.isAwaitable()) {
+            return getSplitSource(table, splitSchedulingStrategy, dynamicFilter);
+        }
+        long blockingTimeoutMillis = getDynamicFilteringProbeBlockingTimeout(session).toMillis();
+        if (blockingTimeoutMillis <= 0) {
+            return getSplitSource(table, splitSchedulingStrategy, dynamicFilter);
+        }
+        CompletableFuture<ConnectorSplitSource> future = dynamicFilter.isBlocked()
+            .completeOnTimeout(null, blockingTimeoutMillis, TimeUnit.MILLISECONDS)
+            .thenApplyAsync(o -> getSplitSource(table, splitSchedulingStrategy, dynamicFilter), executor);
+        return new AsyncDelegate(future, executor);
```

Comment:
I think you can use `directExecutor`, unless you ran into some issue which led to using thread pool here ?

Reply:
Modified